code/handlers/trackservers.q

```diff
@@ -335,10 +335,11 @@ reqprocsnotconn:{[requiredprocs]
   }
 
 // Block process until all required processes are connected
-startupdependent:{[requiredprocs;timeintv]
-    while[.servers.reqprocsnotconn[requiredprocs];
-          .os.sleep[timeintv];
-          .servers.startup[]
+startupdependent:{[requiredprocs;timeintv;cycles;subscribe]
+     a:();
+     while[.servers.reqprocsnotconn[requiredprocs];
+          [a+:timeintv;if[cycles<`int$a%timeintv;.lg.e[`connectionreport;"RDB cannot connect to tickerplant"];exit 1];.os.sleep[timeintv]];
```

Comment:
would it not be simpler to just increment the counter `a` by 1 each time and check if `a>cycles`? Think this line is also a bit long, and there shouldn't need to be `[...]` around this or the next line